lib/ansible/playbook/role/__init__.py

```diff
@@ -376,7 +376,10 @@ def serialize(self, include_deps=True):
 
         parents = []
         for parent in self._parents:
-            parents.append(parent.serialize(include_deps=False))
+            # include parent only once
+            # Fix https://github.com/ansible/ansible/issues/14112
+            if parent not in self._parents:
```

Comment:
Should this be `if parent not in parents:` instead? Otherwise, I do not see how this helps anything.